### ISSUE TITLE
"Cannot parse boolean from string: on" when accessing user account

### ISSUE THREAD

**Issue by konstan** [slipstream/SlipStreamUI#221]
Appears on BB when accessing user/sixsq 
```
>>>> Uncaught exception during the UI generation (clj side):

java.lang.IllegalArgumentException: Cannot parse boolean from string: on
        at slipstream.ui.util.clojure$parse_boolean.invoke(clojure.clj:176)
        at clojure.lang.AFn.applyToHelper(AFn.java:161)
        at clojure.lang.AFn.applyTo(AFn.java:151)
        at clojure.core$apply.invoke(core.clj:619)
        at clojure.core$update_in.doInvoke(core.clj:5587)
        at clojure.lang.RestFn.invoke(RestFn.java:445)
        at slipstream.ui.util.core$normalize_value.invoke(core.clj:146)
        at slipstream.ui.models.parameters$parse_parameter.invoke(parameters.clj:44)
        at clojure.core$map$fn__4207.invoke(core.clj:2487)
        at clojure.lang.LazySeq.sval(LazySeq.java:42)
        at clojure.lang.LazySeq.seq(LazySeq.java:60)
        at clojure.lang.Cons.next(Cons.java:39)
        at clojure.lang.RT.next(RT.java:598)
        at clojure.core$next.invoke(core.clj:64)
        at clojure.core.protocols$fn__6034.invoke(protocols.clj:146)
        at clojure.core.protocols$fn__6005$G__6000__6014.invoke(protocols.clj:19)
        at clojure.core.protocols$seq_reduce.invoke(protocols.clj:31)
        at clojure.core.protocols$fn__6026.invoke(protocols.clj:54)
        at clojure.core.protocols$fn__5979$G__5974__5992.invoke(protocols.clj:13)
        at clojure.core$reduce.invoke(core.clj:6177)
        at clojure.core$group_by.invoke(core.clj:6492)
        at slipstream.ui.util.clojure$coll_grouped_by.doInvoke(clojure.clj:253)
        at clojure.lang.RestFn.invoke(RestFn.java:636)
        at slipstream.ui.models.parameters$group.invoke(parameters.clj:55)
        at slipstream.ui.models.parameters$parameter_categories.invoke(parameters.clj:59)
        at slipstream.ui.models.parameters$parse.invoke(parameters.clj:68)
        at slipstream.ui.views.user$page.invoke(user.clj:47)
        at clojure.lang.AFn.applyToHelper(AFn.java:161)
        at clojure.lang.AFn.applyTo(AFn.java:151)
        at clojure.core$apply.invoke(core.clj:617)
        at slipstream.ui.views.representation$render_html.doInvoke(representation.clj:72)
        at clojure.lang.RestFn.invoke(RestFn.java:423)
        at slipstream.ui.views.representation$render_page.invoke(representation.clj:78)
        at slipstream.ui.views.representation$_toHtml.invoke(representation.clj:113)
        at slipstream.ui.views.Representation.toHtml(Unknown Source)
        at com.sixsq.slipstream.util.HtmlUtil.toHtml(HtmlUtil.java:61)
        at com.sixsq.slipstream.util.HtmlUtil.toHtml(HtmlUtil.java:43)
        at com.sixsq.slipstream.resource.ParameterizedResource.toHtml(ParameterizedResource.java:231)
        at com.sixsq.slipstream.user.UserResource.toHtml(UserResource.java:158)
...        
        at org.restlet.routing.Router.doHandle(Router.java:431)
        at com.sixsq.slipstream.application.RootApplication$RootRouter.doHandle(RootApplication.java:424)
        at org.restlet.routing.Router.handle(Router.java:648)
...
```

![screen shot 2015-03-10 at 11 13 01](https://cloud.githubusercontent.com/assets/353767/6573173/a6b81714-c718-11e4-993c-5becabb77539.png)

**Comment by rbf** [comment]
Working on that.

@schaubl 
This seems related to [`SlipStream/SlipStreamServer/commit/081d1d5`](https://github.com/slipstream/SlipStreamServer/commit/081d1d5aeb1b38705a2197387c795eb5ae56fe78). This that change only how booleans are interpreted from the AJAX requests in the java side, or also how booleans are persisted in the DB?

**Comment by konstan** [comment]
It come from "General.On Error Run Forever" parameter set to 'on'.
```
SELECT category,name,value FROM "PUBLIC"."USERPARAMETER" WHERE "CONTAINER_RESOURCEURI" ='user/sixsq' and value='on'

CATEGORY NAME                         VALUE
-------- ---------------------------- -----
General  General.On Error Run Forever on   
```

**Comment by schaubl** [comment]
The clojure function should do the same as the java implementation here: https://github.com/slipstream/SlipStreamServer/blob/081d1d5aeb1b38705a2197387c795eb5ae56fe78/jar-service/src/main/java/com/sixsq/slipstream/resource/BaseResource.java#L221

**Comment by rbf** [comment]
The `sixsq` user profile was fixed by manually going into `edit` mode (by adding `?edit=true` in the url) and re-saving the profile. This correctly updated the field to `true` and the page renders properly now.

On top of that, to prevent this happening again, we made the [boolean-parsing function in the clj side](https://github.com/slipstream/SlipStreamUI/blob/master/clj/src/slipstream/ui/util/clojure.clj#L162-L181) tolerant to the `on/off`and `yes/no` variations, in the same way that the java code pointed out by the comment above.
